File: app/src/main/java/edu/schoolapp/AssignmentAdapter.java
package edu.schoolapp;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import edu.schoolapp.Model.AssignmentModel;

public class AssignmentAdapter extends RecyclerView.Adapter<AssignmentAdapter.AssignmentVH> {

    private List<AssignmentModel> assignmentModelList;

    public AssignmentAdapter(List<AssignmentModel> list) {
        assignmentModelList = list;
    }

    @NonNull
    @Override
    public AssignmentVH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_assignment, parent, false);
        return new AssignmentVH(view);
    }

    @Override
    public void onBindViewHolder(@NonNull AssignmentVH holder, int position) {
        holder.setupData(assignmentModelList.get(position));
    }

    @Override
    public int getItemCount() {
        return assignmentModelList == null ? 0 : assignmentModelList.size();
    }

    public void updateData(List<AssignmentModel> list) {
        if (assignmentModelList == null) {
            assignmentModelList = new ArrayList<>();
        }
        assignmentModelList.clear();
        assignmentModelList.addAll(list);
        notifyDataSetChanged();
    }

    static class AssignmentVH extends RecyclerView.ViewHolder {

        private TextView titleTextView;
        private TextView unitTextView;
        private TextView descriptionTextView;
        private TextView markTextView;

        AssignmentVH(View itemView) {
            super(itemView);

            titleTextView = itemView.findViewById(R.id.title_text_view);
            unitTextView = itemView.findViewById(R.id.subject_text_view);
            descriptionTextView = itemView.findViewById(R.id.description_text_view);
            markTextView = itemView.findViewById(R.id.mark_text_view);
        }

        void setupData(AssignmentModel item) {
            titleTextView.setText(item.getTitle());
            unitTextView.setText(item.getUnit());
            descriptionTextView.setText(item.getDescription());
            markTextView.setText(String.format("%d out of %d", item.getMarkObtained(), item.getMark()));
        }
    }
}

File: app/src/main/java/edu/schoolapp/AssignmentAdapter.kt
package edu.schoolapp

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import edu.schoolapp.AssignmentAdapter.AssignmentVH
import edu.schoolapp.model.AssignmentModel

class AssignmentAdapter(private var assignmentModelList: MutableList<AssignmentModel>?) :
    RecyclerView.Adapter<AssignmentVH>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AssignmentVH {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_assignment, parent, false)
        return AssignmentVH(view)
    }

    override fun onBindViewHolder(holder: AssignmentVH, position: Int) {
        holder.setupData(assignmentModelList!![position])
    }

    override fun getItemCount(): Int {
        return if (assignmentModelList == null) 0 else assignmentModelList!!.size
    }

    fun updateData(list: List<AssignmentModel>?) {
        if (assignmentModelList == null) {
            assignmentModelList = ArrayList()
        }
        assignmentModelList!!.clear()
        assignmentModelList!!.addAll(list!!)
        notifyDataSetChanged()
    }

    class AssignmentVH(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val titleTextView: TextView = itemView.findViewById(R.id.title_text_view)
        private val unitTextView: TextView = itemView.findViewById(R.id.subject_text_view)
        private val descriptionTextView: TextView = itemView.findViewById(R.id.description_text_view)
        private val markTextView: TextView = itemView.findViewById(R.id.mark_text_view)

        fun setupData(item: AssignmentModel) {
            titleTextView.text = item.title
            unitTextView.text = item.unit
            descriptionTextView.text = item.description
            markTextView.text = String.format("%d out of %d", item.markObtained, item.mark)
        }
    }
}